packages/ui-dialog/src/Dialog/props.ts

```diff
@@ -28,30 +28,34 @@ import PropTypes from 'prop-types'
 import type {
   AsElementType,
   OtherHTMLAttributes,
-  PropValidators
+  PropValidators,
+  UIElement
 } from '@instructure/shared-types'
+import type { FocusRegionOptions } from '@instructure/ui-a11y-utils'
 
 type DialogOwnProps = {
+  /**
+   * The children to be rendered within the `<Dialog />`
+   */
   children?: React.ReactNode
+  /**
+   * The element to render as the component root, `span` by default
+   */
   as?: AsElementType
   display?: 'auto' | 'block' | 'inline-block'
+  /**
+   * The aria-label to read for screen reader.
+   */
   label?: string
+  /**
+   * Whether or not the `<Dialog />` is open
+   */
   open?: boolean
-  onBlur?: (...args: any[]) => any
-  onDismiss?: (...args: any[]) => any
-  defaultFocusElement?: React.ReactElement | ((...args: any[]) => any)
-  contentElement?: React.ReactElement | ((...args: any[]) => any)
-  liveRegion?:
-    | React.ReactElement[]
-    | React.ReactElement
-    | ((...args: any[]) => any)
-  shouldContainFocus?: boolean | ('keyboard' | 'screenreader')
-  shouldReturnFocus?: boolean
-  shouldCloseOnDocumentClick?: boolean
-  shouldCloseOnEscape?: boolean
-  shouldFocusOnOpen?: boolean
-  elementRef?: (element: Element | null) => void
-}
+  /**
+   * An element or a function returning an element that wraps the content of the `<Dialog />`
+   */
+  contentElement?: UIElement
+} & FocusRegionOptions
```

Comment:
I like that this is used from FocusRegion, but the problem is, that the current doc generation cannot read the comments from imported files, so it doesn't appear on Dialogs Properties block. For a workaround (until this is fixed somehow) I suggest listing the props like this: 
```js
...
  /**
   * An element, function returning an element, or array of elements that will not be hidden from
   * the screen reader when the `<Dialog />` is open
   */
  liveRegion?: FocusRegionOptions['liveRegion']

  /**
   * Function called when tab focus leaves the `<Dialog />` focusable content. This only
   * occurs when `shouldContainFocus` is set to false.
   */
  onBlur?: FocusRegionOptions['onBlur']
...
```
etc...

What do you think?

Reply:
hmm I'm not a fan of it, we're duplicating code because of a buggy tool :/ Also its not making things much better, because `LiveRegion` is in `shared-types`. 
And the code documentation is already readable in the IDE autocomplete popup.

Reply:
Then another idea: if you put back the original comments for those props in the propTypes block, the doc generation can pick those up, and we don't have to touch the types.

Reply:
OK, did this, the comments are now showing up.